students/navdeep/session07/html_render.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+
+"""
+A class-based system for rendering html.
+"""
+
+# This is the framework for the base class
+class Element(object):
+    open_tag = ''
+    close_tag = ''
+    indentation = "    "
+    def __init__(self, content=None):
+        self.content = []
+        if content:
+            self.content.append(content)
+
+    def append(self, new_content):
+        self.content.append(new_content)
+
+    def render(self, file_out, cur_ind=""):
+        format_html = ""
+        for html_content in self.content:
+            if type(html_content) == str:
+                format_html = cur_ind + self.open_tag + '\n' + (cur_ind + "  ") + html_content +'\n' + cur_ind + self.close_tag +'\n'
+            else:
+                html_content.render(file_out, cur_ind + self.indentation)
+            file_out.write(format_html)
+
+class Html(Element):
+    def __init__(self, content = None):
+        self.open_tag = "<html>"
+        self.close_tag = "</html>"
```

Comment:
You could also just set tag = "html" and then set up each element to automatically add <> and </> to the beginning and ending tags to reduce redundancy.